app/toolbox-server.hs

```diff
@@ -0,0 +1,93 @@
+module Main (main) where
+
+import Concur.Core
+  ( Widget,
+    liftSTM,
+  )
+import Concur.Replica
+  ( div,
+    pre,
+    runDefault,
+    text,
+  )
+import Control.Applicative ((<|>))
+import Control.Concurrent (forkIO)
+import Control.Concurrent.STM
+  ( TVar,
+    atomically,
+    newTVar,
+    readTVar,
+    retry,
+    writeTVar,
+  )
+import Data.Map (Map)
+import qualified Data.Map as M
+import Data.Text (Text)
+import qualified Data.Text as T
+import Replica.VDOM.Types (HTML)
+import Toolbox.Comm
+  ( receiveObject,
+    runServer,
+  )
+import Prelude hiding (div)
+
+main :: IO ()
+main = do
+  var <- atomically $ newTVar Nothing
+  _ <- forkIO $ listener var
+  webServer var
+
+type ModuleMessages = Map Text Text
+
+listener :: TVar (Maybe (Int, ModuleMessages)) -> IO ()
+listener var =
+  runServer
+    "/tmp/ghc-build-analyzer.ipc"
```

Comment:
May want to use [`temporary`](https://www.stackage.org/haddock/lts-19.16/temporary-1.3/System-IO-Temp.html) or [`unix`](https://www.stackage.org/haddock/lts-19.16/unix-2.7.2.2/System-Posix-Temp.html) for these

Or switch to an HTTP server, that way we don't have to reinvent the wheel so much here

Reply:
Yes, I will use `temporary`. As for http server, I avoided that simply not to add more dependencies to plugin (as plugin will add its deps to the target code) and HTTP port is a rather scarce resource. However, as we can reuse library codes and open up new use cases with remote monitoring interface, HTTP interface may be a better choice. I will switch this to that in a later PR at some point.

Reply:
I just found that the automatic generation of temporary directory and file from `temporary` would not fit to the purpose as plugin (ghc) and the server (toolbox-server) are separate processes. We had better pass the shared IPC socket file path or config file explicitly. We can pass plugin argument to GHC by the `-fplugin-opt` flag. I will make a separate PR for that change.

Reply:
Now created PR #9. Both toolbox-server and Plugin.CheckImports take the socket file path as a CLI argument and -fplugin-opt argument, respectively.